File: programmers/js/basic/181929.js
// 원소들의 곱과 합
function solution(num_list) {
  return num_list.reduce((acc, cur) => acc * cur, 1) <
    num_list.reduce((acc, cur) => acc + cur, 0) ** 2
    ? 1
    : 0;
}

// 다른 풀이
// function solution(num_list) {
//   let accMul = 1;
//   let accSum = 0;

//   for (const num of num_list) {
//     accMul *= num;
//     accSum += num;
//   }

//   return accMul < accSum ** 2 ? 1 : 0;
// }

File: programmers/java/basic/181929.java
// 원소들의 곱과 합

class Solution {
    public int solution(int[] num_list) {
        int num1 = 1;
        int num2 = 0;
        
        for (int num : num_list) {
            num1 *= num;
            num2 += num;
        }
        
        return num1 < num2 * num2 ? 1 : 0;
    }
}